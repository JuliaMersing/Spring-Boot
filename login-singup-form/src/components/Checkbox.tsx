import React, { FormEvent } from 'react';

type CheckboxProps = {
  onChange: (e: FormEvent) => any,
  check: boolean,
}

const Checkbox = ({
  onChange,
  check,
}: CheckboxProps) => (
  <input
    type="checkbox"
    className="checkbox"
    defaultChecked={check}
    onChange={onChange}
  />
);

export default Checkbox;
