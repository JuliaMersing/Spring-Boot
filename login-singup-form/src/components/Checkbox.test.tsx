import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Checkbox from './Checkbox';

describe('Checkbox component test', () => {
  const setUp = () => {
    const mockHandleRememberChange = jest.fn();

    const utils = render(
      <Checkbox
        onChange={mockHandleRememberChange}
        check={false}
      />,
    );
    return {
      ...utils,
      mockHandleRememberChange,
    };
  };
  test('should match snapshot', () => {
    setUp();
    expect(screen.getByRole('checkbox'))
      .toMatchSnapshot();
  });
  test('should call remember change', async () => {
    const {
      mockHandleRememberChange,
      getByRole,
    } = setUp();

    const remember = await getByRole('checkbox');
    await userEvent.click(remember);
    expect(mockHandleRememberChange)
      .toHaveBeenCalledTimes(1);
  });
  test(
    'should call  rememberCheck when remember checkBox is checked',
    async () => {
      const {
        getByRole,
      } = setUp();

      const remember = await getByRole('checkbox') as HTMLInputElement;
      await userEvent.click(remember);
      expect(remember.checked)
        .toEqual(true);

      await userEvent.click(remember);
      expect(remember.checked)
        .toEqual(false);
    },
  );
});
