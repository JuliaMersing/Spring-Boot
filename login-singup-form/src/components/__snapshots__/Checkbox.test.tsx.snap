// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component test should match snapshot 1`] = `
<input
  class="checkbox"
  type="checkbox"
/>
`;
